app/codec/ffmpeg/ffmpegencoder.cpp

```diff
@@ -721,6 +721,7 @@ bool FFmpegEncoder::SetupCodecContext(AVStream* stream, AVCodecContext* codec_ct
 
   if (codec->type == AVMEDIA_TYPE_VIDEO) {
     stream->avg_frame_rate = codec_ctx->framerate;
+    stream->time_base = av_add_q(codec_ctx->time_base, {0, 1});
```

Comment:
What does this do? Is this to ensure a non-zero denominator?

Reply:
oh no. That wasn't supposed to make it in, Ill change this back to draft. 

but explaining what it is, and what we need it for, we tell the encoder what framerate to use but when writing to an mxf, it also needs to be explicitly told what time_base it's being told to use. this is so that it can ensure interoperability between applications. (one of the reasons people love mxf so much)

just setting `stream->time_base = codec_ctx->time_base` will make any non-supported framerates error out. which isn't a massive issue since nearly all the video framerates that are exposed by default work fine. with the exception of 10fps and 15fps. so this was never supposed to make it in. 

what WAS supposed to make it in, was a better error message explaining why the encode failed immediately instead of error that pops up now "failed to open file: failed to write format header: Invalid argument -22"

the {0, 1} was debugging crap that was supposed to be removed. but I pushed and old commit by accident. but I suppose this is a good time to ask, would it be better to just report why it failed, or find a way to just show the supported framerates? 

I had done a fresh install since the time I worked on this so I lost the code either way. and while I don't think anybody who will export DNxHD will try to export 15 or 10 fps I'm not sure if the error message it currently serves is good enough or not.

Reply:
I think an error message should be sufficient, you can see a similar use case here to inform users that H.264 and H.265 require dimensions to be a multiple of 2: https://github.com/olive-editor/olive/blob/master/app/dialog/export/export.cpp#L357

I think using a similar approach for this limitation would be perfectly fine